Sorting/RadixSort.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+    Radix sort
+    ----------
+    An efficient, key-index-based, comparision based sorting algorithm.
+    It sorts based on the keys obtained on each index of the numbers to be sorteed.
+*/
+
+#include <iostream>
+#include <vector>
+#include <array>
+#include <algorithm>
+#include "SortingUtils.hpp"
+
+using namespace std;
+
+/*
+    count_sort
+    ----------
+    Performs Count Sort on the elements passed by the radix_sort function
+
+    Time complexity
+    ---------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    Hence, the above expression becomes O(N + 10) = O(N)
+
+    Space complexity
+    ----------------
+    O(N + B), where N is the number of keys and B is the base for representing the numbers(10 in case of decimal system)
+    Hence, the above expression becomes O(N + 10) = O(N)
+    This space is used for count sorting each key and temporarily creating a sorted array.
+*/
+
+auto counter_accessor = [] (const int value, const int mult_factor, const int add_factor, const int extractor) {
+    return add_factor + (mult_factor * ((value/extractor) % 10));
+};
```

Comment:
Have this local to `count_sort()` so that you wouldn't need to pass `mult_factor`, `add_factor` and `extractor` as arguments, they will automatically be captured as references if you declare the lambda as in the example I gave.